### PR TITLE
object endpoint to find internal links for resources

### PR DESCRIPTION
**Pull request by recalcitrantsupplant**
From dev readme:
**High Level Sequence `/object` endpoint**

Prez provides a `/object` endpoint as an endpoint that supplies any information known about a given URI. If an annotated
mediatype is requested, prez will additionally provide all system links for endpoints which can render the object. The
high level sequence for this endpoint is as follows:

1. Get the URI for the object from the query string
2. Get the class(es) of the object from the triplestore
3. Use prez's reference data for endpoints to determine which endpoints can render this object, and, a template for
these endpoints, specifying any variables that need to be substituted (such as parent URIs).
4. Get the object information from the triplestore, using an open profile, and in parallel any system information needed
to construct the system links.
5. Return the response
---
The same functions that are used to do the above have been generally applied to annotated responses. Any annotated response from prez will now:
1. Find any URIs in the initial response from the triplestore
2. Find the classes for these URIs
3. Use the (new) prez endpoints graph (`reference_data/endpoints/`) to find endpoints that can deliver the class(es), and, get a template for these endpoints, specifying any variables that need to be substituted (such as parent URIs).
4. Go back to the triplestore to populate the endpoint templates.
5. Return the original response + prez:links
---
**Other minor changes:**
- Add endpoint reference data to support system link generation (`reference_data/endpoints`)
- Rename inbound/outbound children/parents to focus to/from child/parent terminology to align with SHACL. Rename general_class to base_class. Resolves https://github.com/RDFLib/prez/issues/101
- Add a triple: of the form: ~~`<focus_uri> prez:endpointComponentURI <uri_used_in_endpoint>`~~ `<endpoint_component_uri> dcterms:identifier "{curie_id}"^^prez:identifier` for parent CURIEs in endpoints such that the URIs of parents are included in the response, and get annotated, such that labels are available for breadcrumbs.

### CODE REVIEW

**Review by jamiefeiss — CHANGES_REQUESTED** [review]
Tested locally, the `/profiles` endpoint doesn't return `prez:links`

**Comment by edmondchuc** [comment]
Can confirm. I have the same issue as @jamiefeiss.

**Comment by recalcitrantsupplant** [comment]
I have a fix coming, should be tomorrow.

**Review by jamiefeiss — CHANGES_REQUESTED** [review]
After some local testing, I've found two issues:

1. For a `skos:Concept` page, the `prez:endpointComponentURI` triple seems to be missing:

```turtle
aarr:chair a skos:Concept ;
    dcterms:identifier "aarr:chair"^^prez1:identifier ;
    dcterms:provenance "Developed for this Agent to Agent Roles vocabulary"@en ;
    rdfs:isDefinedBy <https://data.idnau.org/pid/vocab/aarr> ;
    skos:broader aarr:member ;
    skos:definition "The presiding officer of an assembly, meeting, committee, or board"@en ;
    skos:inScheme <https://data.idnau.org/pid/vocab/aarr> ;
    skos:prefLabel "Chair"@en ;
    skos:scopeNote "Organisation to Person relations" ;
    prez1:link "/v/vocab/vcb:aarr/aarr:chair" .
```

2. For a `skos:Collection` page,  a server error occurs: `AttributeError: 'list' object has no attribute 'name'`, which points to this line:

https://github.com/RDFLib/prez/blob/4d5a0ba0bdcae67323d2373fed655c2aa05e6697/prez/services/generate_profiles.py#L90

where the problem seems to be a missing class in the SPARQL query:
```sparql
VALUES ?class {}
```

**Comment by recalcitrantsupplant** [comment]
Thanks Jamie I'll take a look.

**Comment by recalcitrantsupplant** [comment]
@jamiefeiss both fixed now - note I've switched to the dcterms:identifier to include parent URL components / their labels. From the updated notes:

_Add a triple: of the form: `<endpoint_component_uri> dcterms:identifier "{curie_id}"^^prez:identifier` for parent CURIEs in endpoints such that the URIs of parents are included in the response, and get annotated, such that labels are available for breadcrumbs._

**Review by edmondchuc — CHANGES_REQUESTED** [review]

File: prez/cache.py [hunk 1]
```diff
@@ -5,6 +5,9 @@
 profiles_graph_cache = ConjunctiveGraph()
 profiles_graph_cache.bind("prez", "https://prez.dev/")
 
+endpoints_graph_cache = ConjunctiveGraph()
+profiles_graph_cache.bind("prez", "https://prez.dev/")
```
[the remark this answers is not in the document]
Comment (recalcitrantsupplant) [reply prez/cache.py]: I wonder if in future we change all of the system graphs to an RDFLib Dataset

[the remark this answers is not in the document]
Comment (edmondchuc) [reply prez/cache.py]: Yes we should migrate over soon. ConjunctiveGraph is marked for removal in a future release of RDFLib.

**Review by jamiefeiss — APPROVED** [review]
The frontend has everything it needs now after local testing. Thanks David

**Comment by recalcitrantsupplant** [comment]
@edmondchuc I think I've addressed the feedback (just the single suggested change) let me know if I've missed something otherwise will merge

**Review by jamiefeiss — CHANGES_REQUESTED** [review]
I found another issue I must've overlooked before: search results are missing labels and geometries. See the screenshot below, the map with geometries should be displaying on the right:

image

**Comment by recalcitrantsupplant** [comment]
thanks Jamie, not sure if these changes caused that issue, in any case I'd like to merge this in before this PR gets any larger. Could you raise an issue separately? I'll address it when making the filter addition to search.

**Pull request merged**